assets/js/main.js

```diff
@@ -147,75 +147,97 @@ $('.progress-bar').each(function(index) {
+    this.value = Math.min(this.value, maxThumb.value - 2);
+    var totalSteps = parseInt(this.max) - parseInt(this.min);
+    var percent = (100 / totalSteps) * parseInt(this.value) - (100 / totalSteps) * parseInt(this.min);
+    var $container = $(this).parent();
+    $container.find("div[inverse-left]").css("width", percent + '%');
+    $container.find("div[range]:first,div[sign-one],span[thumb]:first").css("left", percent + '%');
+    $container.find("div[sign-one] span").text(this.value + label);
+  });
 
-    $(element2).on("input",function(){
-      var label = $(this).attr("data-label");
-      var minThumb = $(this).parent().find(".thumb-1")[0];
-      this.value=Math.max(this.value,minThumb.value-(-2));
-      var totalSteps = parseInt(this.max)-parseInt(this.min);
-      var percent=(100/totalSteps)*parseInt(this.value)-(100/totalSteps)*parseInt(this.min);
-      var $container = $(this).parent();
-      $container.find("div[inverse-right]").css("width",(100 - percent) +'%');
-      $container.find("div[range]:last").css("right",(100-percent)+'%')
-      $container.find("div[sign-two],span[thumb]:last").css("left", percent+'%');
-      $container.find("div[sign-two] span").text(this.value + label);
-    });
-  };
+  $(element2).on("input", function () {
+    var label = $(this).attr("data-label");
+    var minThumb = $(this).parent().find(".thumb-1")[0];
+    this.value = Math.max(this.value, minThumb.value - (-2));
+    var totalSteps = parseInt(this.max) - parseInt(this.min);
+    var percent = (100 / totalSteps) * parseInt(this.value) - (100 / totalSteps) * parseInt(this.min);
+    var $container = $(this).parent();
+    $container.find("div[inverse-right]").css("width", (100 - percent) + '%');
+    $container.find("div[range]:last").css("right", (100 - percent) + '%')
+    $container.find("div[sign-two],span[thumb]:last").css("left", percent + '%');
+    $container.find("div[sign-two] span").text(this.value + label);
+  });
+};
 
-  $('.dp-show-tips').on('click', function(){
-    $(this).closest('.field-item').next('.dp-advice').slideToggle();
+$('.dp-show-tips').on('click', function () {
+  $(this).closest('.field-item').next('.dp-advice').slideToggle();
 
-    var $showPanelLink = $(this).find('.dp-show-text');
-    if ($(this).find('.icon-close').hasClass('rotation')) {
-      $showPanelLink.html($(this).attr('data-text-hide'));
-      } else {
-        $showPanelLink.html($(this).attr('data-text-show'));
-    }
-    $(this).find('.icon-close').toggleClass('rotation');
+  var $showPanelLink = $(this).find('.dp-show-text');
+  if ($(this).find('.icon-close').hasClass('rotation')) {
+    $showPanelLink.html($(this).attr('data-text-hide'));
+  } else {
+    $showPanelLink.html($(this).attr('data-text-show'));
+  }
+  $(this).find('.icon-close').toggleClass('rotation');
 
-  });
+});
 
+// Carousel
+
+(function () {
+  $(document).ready(function () {
+    return $("input[type='radio']").change(function () {
```

Comment:
This will affect all apps using the UI library, It would be best like this:
```suggestion
    return $("input[type='radio'][name='carousel1'], input[type='radio'][name='carousel2'] ").change(function () {
```
To target only what we have in the ui library.

Reply:
I get it. It happens that in that case, if you want to add another carousel, you have to touch the js of the library. Ex, to put another carousel, "carousel3"

Reply:
Since it is calling the slides whose id ends in `_dpslide`, I think it would not affect all the inputs

Reply:
You can try with a console.log, each input type radio will trigger this event. Even if it does nothing but to `_dpslide` ending divs, it is still code running that has not to be in the event. 👀

Reply:
The idea here is that you either use the js in the library or don't.

Reply:
This could be replaced by this:
```
 $("input[type='radio'].dp-carousel-dot")
 ```